src/pybind/mgr/dashboard/frontend/src/app/core/auth/login/login.component.spec.ts

```diff
@@ -29,4 +40,14 @@ describe('LoginComponent', () => {
     component.ngOnInit();
     expect(component['modalService'].hasOpenModals()).toBeFalsy();
   });
+
+  it('should not show create cluster wizard if cluster creation was successful', () => {
```

Comment:
Here you're not testing the component `login` method. This is a refactoring on how to test the login method:
```
diff --git a/src/pybind/mgr/dashboard/frontend/src/app/core/auth/login/login.component.spec.ts b/src/pybind/mgr/dashboard/frontend/src/app/core/auth/login/login.component.spec.ts
index 808b299f7be..9fdad0807e2 100644
--- a/src/pybind/mgr/dashboard/frontend/src/app/core/auth/login/login.component.spec.ts
+++ b/src/pybind/mgr/dashboard/frontend/src/app/core/auth/login/login.component.spec.ts
@@ -1,31 +1,31 @@
 import { HttpClientTestingModule } from '@angular/common/http/testing';
 import { ComponentFixture, TestBed } from '@angular/core/testing';
 import { RouterTestingModule } from '@angular/router/testing';
+import { Router } from '@angular/router';
 
+import { of } from 'rxjs';
+
+import { AuthService } from '~/app/shared/api/auth.service';
 import { configureTestBed } from '~/testing/unit-test-helper';
 import { AuthModule } from '../auth.module';
 import { LoginComponent } from './login.component';
 
-class MockUrlGenerator {
-  url = '/';
-  showClusterCreationPage(postInstalled: boolean) {
-    if (!postInstalled) {
-      this.url = '/create-cluster';
-    }
-  }
-}
-
 describe('LoginComponent', () => {
   let component: LoginComponent;
   let fixture: ComponentFixture<LoginComponent>;
-  let urlGenerator: MockUrlGenerator;
+  let routerNavigateSpy: jasmine.Spy;
+  let authServiceLoginSpy: jasmine.Spy;
 
   configureTestBed({
     imports: [RouterTestingModule, HttpClientTestingModule, AuthModule]
   });
 
   beforeEach(() => {
-    urlGenerator = new MockUrlGenerator();
+    routerNavigateSpy = spyOn(TestBed.inject(Router), 'navigate');
+    routerNavigateSpy.and.returnValue(true);
+    authServiceLoginSpy = spyOn(TestBed.inject(AuthService), 'login');
+    authServiceLoginSpy.and.returnValue(of(null));
+
     fixture = TestBed.createComponent(LoginComponent);
     component = fixture.componentInstance;
     fixture.detectChanges();
@@ -38,16 +38,23 @@ describe('LoginComponent', () => {
   it('should ensure no modal dialogs are opened', () => {
     component['modalService']['modalsCount'] = 2;
     component.ngOnInit();
+
     expect(component['modalService'].hasOpenModals()).toBeFalsy();
   });
 
-  it('should not show create cluster wizard if cluster creation was successful', () => {
-    urlGenerator.showClusterCreationPage(true);
-    expect(urlGenerator.url).toBe('/');
+  it('should not show create cluster wizard if cluster creation is completed', () => {
+    component.postInstalled = true;
+    component.login();
+
+    expect(routerNavigateSpy).toHaveBeenCalledTimes(1);
+    expect(routerNavigateSpy).toHaveBeenCalledWith(['/']);
   });
 
-  it('should show create cluster wizard if cluster creation was failed', () => {
-    urlGenerator.showClusterCreationPage(false);
-    expect(urlGenerator.url).toBe('/create-cluster');
+  it('should show create cluster wizard if cluster creation is not completed', () => {
+    component.postInstalled = false;
+    component.login();
+
+    expect(routerNavigateSpy).toHaveBeenCalledTimes(1);
+    expect(routerNavigateSpy).toHaveBeenCalledWith(['/create-cluster']);
   });
 });
diff --git a/src/pybind/mgr/dashboard/frontend/src/app/core/auth/login/login.component.ts b/src/pybind/mgr/dashboard/frontend/src/app/core/auth/login/login.component.ts
index 31f3cb463e7..42876695789 100644
--- a/src/pybind/mgr/dashboard/frontend/src/app/core/auth/login/login.component.ts
+++ b/src/pybind/mgr/dashboard/frontend/src/app/core/auth/login/login.component.ts
@@ -65,10 +65,8 @@ export class LoginComponent implements OnInit {
 
   login() {
     this.authService.login(this.model).subscribe(() => {
-      let url = _.get(this.route.snapshot.queryParams, 'returnUrl', '/');
-      if (!this.postInstalled) {
-        url = _.get(this.route.snapshot.queryParams, 'returnUrl', '/create-cluster');
-      }
+      let urlPath = this.postInstalled ? '/' : '/create-cluster';
+      let url = _.get(this.route.snapshot.queryParams, 'returnUrl', urlPath);
       this.router.navigate([url]);
     });
   }
```

Reply:
Thank you! Done and updated.